From 76f74da80e53d63e949b1a27e827e7106463f6bb Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Tue, 2 Apr 2019 14:42:24 +0530
Subject: [PATCH 132/386] octeontx2-pf: Fix PF driver crash on driver unbind

Free VF FLR/ME and mbox interrupts only if VFs are enabled.

Change-Id: I4404942122f05b17c9859ca307f97ccd3c7c03dd
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/6734
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index cf2790c6920b..87b64cd2def5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -2405,6 +2405,9 @@ static int otx2_sriov_disable(struct pci_dev *pdev)
 	struct otx2_nic *pf = netdev_priv(netdev);
 	int i;
 
+	if (!pci_num_vf(pdev))
+		return 0;
+
 	pci_disable_sriov(pdev);
 
 	for (i = 0; i < pci_num_vf(pdev); i++)
-- 
2.17.1

